.golangci.yml

```diff
@@ -270,114 +570,27 @@ linters-settings:
-
-  gosec:
-    # To select a subset of rules to run.
-    # Available rules: https://github.com/securego/gosec#available-rules
-    # Default: [] - means include all rules
-    # includes:
-    # To specify a set of rules to explicitly exclude.
-    # Available rules: https://github.com/securego/gosec#available-rules
-    # Default: []
-    # excludes:
-    # Exclude generated files
-    # Default: false
-    exclude-generated: false
-    # Filter out the issues with a lower severity than the given value.
-    # Valid options are: low, medium, high.
-    # Default: low
-    severity: medium
-    # Filter out the issues with a lower confidence than the given value.
-    # Valid options are: low, medium, high.
-    # Default: low
-    confidence: medium
-    # Concurrency value.
-    # Default: the number of logical CPUs usable by the current process.
-    concurrency: 12
-
-  whitespace:
-    # Enforces newlines (or comments) after every multi-line if statement.
-    # Default: false
-    multi-if: true
-    # Enforces newlines (or comments) after every multi-line function signature.
-    # Default: false
-    multi-func: true
-
-  wrapcheck:
-    # An array of strings that specify substrings of signatures to ignore.
-    # If this set, it will override the default set of ignored signatures.
-    # See https://github.com/tomarrell/wrapcheck#configuration for more information.
-    # Default: [".Errorf(", "errors.New(", "errors.Unwrap(", ".Wrap(", ".Wrapf(", ".WithMessage(", ".WithMessagef(", ".WithStack("]
-    ignoreSigs:
-      - .Errorf(
-      - errors.New(
-      - .WithMessage(
-      - .WithMessagef(
-      - .WithStack(
+  nakedret:
+    # Make an issue if func has more lines of code than this setting, and it has naked returns.
+    # Default: 30
+    max-func-lines: 30
 
-linters:
-  # Disable all linters.
-  # Default: false
-  disable-all: true
-  # Enable specific linter
-  # https://golangci-lint.run/usage/linters/#enabled-by-default
-  enable:
-    - gci
-    - importas
-    - govet
-    - godot
-    - gofmt
-    - gofumpt
-    - revive
-    - misspell
-    - staticcheck
-    - unused
-    - ineffassign
-    - typecheck
-    - nakedret
-    - depguard
-    - goconst
-    # - whitespace
-    # - wrapcheck
-    # - errcheck
-    # - gosec
-    # - nakedret
-    # - gocyclo
-    # - lll
-    # - prealloc
-  
-  # Enable all available linters.
-  # Default: false
-  enable-all: false
+  revive:
+    # Sets the default failure confidence.
+    # This means that linting errors with less than 0.8 confidence will be ignored.
+    # Default: 0.8
+    confidence: 0.8
 
-  # Disable specific linter
-  # https://golangci-lint.run/usage/linters/#disabled-by-default
-  # disable:
-    
-  # Run only fast linters from enabled linters set (first run won't be fast)
-  # Default: false
-  fast: false
+  staticcheck:
+    # SAxxxx checks in https://staticcheck.io/docs/configuration/options/#checks
+    # Default: ["*"]
+    checks: ["all", "-SA5001"]
```

Comment:
-SA5001 seems new, why are we adding it?

Reply:
This check disallows calling `Close` without checking the returned error: https://staticcheck.dev/docs/checks/#SA5001

Seems like I misunderstood the check. I assumed that this is supposedly how we should handle a deferred close:

```golang
f, err := os.Open("")
defer func() {
    _ = f.Close()
}()
```

But actually it's because `Close` is deferred before `err` is checked. Will remove the exception and instead fix the code that causes the problems.